input/postgres/relations.go

```diff
@@ -18,7 +18,8 @@ const relationsSQLOidField = "c.relhasoids AS relation_has_oids"
 const relationsSQLpg12OidField = "false AS relation_has_oids"
 
 const relationsSQL string = `
-	 WITH locked_relids AS (SELECT DISTINCT relation relid FROM pg_catalog.pg_locks WHERE mode = 'AccessExclusiveLock' AND relation IS NOT NULL AND locktype = 'relation')
+	 WITH locked_relids AS (SELECT DISTINCT relation relid FROM pg_catalog.pg_locks WHERE mode = 'AccessExclusiveLock' AND relation IS NOT NULL AND locktype = 'relation'),
+ partition_children AS (SELECT inhparent, array_agg(inhrelid) AS relids FROM pg_catalog.pg_inherits JOIN pg_catalog.pg_partitioned_table ON (partrelid = inhrelid) GROUP BY 1)
```

Comment:
In the (now removed) mergePartitionSizes code there was a TODO that we should include nested child partitions as well.

This is would now fairly easy to implement by making this a recursive CTE (or at least checking a second nesting level), but I think it may be best to change this behavior in a future PR (or even release, to let this change here settle for one cycle).